src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -81,22 +76,35 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
               IndexLogEntry.schemaString(schema),
               numBuckets)),
           Content(path.toString, Seq()),
-          Source(SparkPlan(sourcePlanProperties), Seq(Hdfs(sourceDataProperties))),
+          Source(
+            SparkPlan(sourcePlanProperties)),
           Map())
 
       case None => throw HyperspaceException("Invalid plan for creating an index.")
     }
   }
 
-  protected def sourceFiles(df: DataFrame): Seq[String] =
+  protected def sourceRelations(df: DataFrame): Seq[Relation] =
     df.queryExecution.optimizedPlan.collect {
       case LogicalRelation(
-          HadoopFsRelation(location: PartitioningAwareFileIndex, _, _, _, _, _),
-          _,
-          _,
-          _) =>
-        location.allFiles.map(_.getPath.toString)
-    }.flatten
+      HadoopFsRelation(location: PartitioningAwareFileIndex, _, dataSchema, _, fileFormat, _),
+      _,
+      _,
+      _) =>
+        val files = location.allFiles.map(_.getPath.toString)
+        // Note: Source files are fingerprinted as part of the serialized logical plan currently.
```

Comment:
Can we update this now that we don't have a serialized plan?
`// Note that source files are currently fingerprinted when the optimized plan is fingerprinted by LogicalPlanFingerprint.`